APIS-827 - Handle Missing Fields from DES

AndyBowes opened the pull request:
Fix to handle missing FirstName, FamilyName, AddressLine1 & AddressLine2 in JSON returned by DES

hughfdjackson commented on app/uk/gov/hmrc/openidconnect/userinfo/services/UserInfoTransformer.scala:
```diff
@@ -51,19 +51,19 @@ trait UserInfoTransformer {
 
   private def formattedAddress(desAddress: DesAddress, country: Option[String]) = {
-    s"${desAddress.line1}\n${desAddress.line2}${addLine(desAddress.line3)}${addLine(desAddress.line4)}${addLine(desAddress.postcode)}${addLine(country)}"
+    s"${desAddress.line1.get}${addLine(desAddress.line2)}${addLine(desAddress.line3)}${addLine(desAddress.line4)}${addLine(desAddress.postcode)}${addLine(country)}"
```
This isn't safe - we'll still get errors thrown when creating a formatted address when there is no line1 or line2 addresses.

It would be better to do: 

```
 +    s"${addLine(desAddress.line1}${addLine(desAddress.line2)}${addLine(desAddress.line3)}${addLine(desAddress.line4)}${addLine(desAddress.postcode)}${addLine(country)}" 
```

Perhaps some 'parsing addresses' unit tests would help with catching this sort of thing?

hughfdjackson replied:
Or rather, rendering addresses from DesUserInfo result unit tests :)

AndyBowes replied:
You are right.  It would fail with a missing line1 although a missing line2 should be fine.
I will look at a better solution to fold a list of Option[String] into the full address & create additional tests.

AndyBowes commented:
Modified & simplified the Address formatting to use flatten over a sequence of Option[String].
Added Unit Tests which inject Addresses with missing elements.
